### PR TITLE
Add Rollup to build packages for distribution

### PR DESCRIPTION
**Pull request by alex-ketch**
⚠️ Needs to be tested across platforms, especially browser to make sure all dependencies are present.

Generates CJS, ESM, Browser packages.

Removed use of `export default …` due to:
Screen Shot 2019-10-16 at 13 11 48

Note: I gave Parcel a shot initially, which works but does not have ability to produce modules of separate files.

### CODE REVIEW

File: src/base/Executor.test.ts [hunk 1]
```diff
@@ -249,14 +249,14 @@ class DeepThought extends Executor {
...
+    })
   }
 
   public async execute(node: Node): Promise<Node> {
     if (isA('CodeChunk', node) && node.text === DeepThought.question) {
-      return { ...node, outputs: [42] }
+      return Promise.resolve({ ...node, outputs: [42] })
     }
-    return node
+    return Promise.resolve(node)
```
Comment (nokome) [inline src/base/Executor.test.ts]: Just out of interest were all these new `Promise.resolve`s added by a newer version of `eslint`?

File: src/base/Server.ts [hunk 2]
```diff
@@ -120,20 +120,26 @@ export default abstract class Server {
   /**
    * Start the server
    */
-  public async start(): Promise<void> {}
+  public async start(): Promise<void> {
+    // TODO: Implement function
```
Comment (nokome) [inline src/base/Server.ts]: In this base class, there is nothing to do here - we are leaving it for derived classes to override. So, we should either remove this TODO comment, or if it is less confusing, make this function abstract.

File: src/base/Executor.test.ts (hunk 1 above)
Comment (nokome) [inline src/base/Executor.test.ts]: Out of interest, are all these `Promise.resolve`s added by new version of `eslint`

File: src/base/Server.ts [hunk 3]
```diff
@@ -120,20 +120,26 @@ export default abstract class Server {
...
+    return Promise.resolve()
```
Comment (nokome) [inline src/base/Server.ts]: There is nothing to do here. So either remove the TODO, or if it is less confusing, make this method `abstract`.

File: src/base/Server.ts [hunk 4]
```diff
@@ -120,20 +120,26 @@ export default abstract class Server {
...
+  }
 
   /**
    * Stop the server
    */
-  public async stop(): Promise<void> {}
+  public async stop(): Promise<void> {
+    // TODO: Implement function
+    return Promise.resolve()
```
Comment (nokome) [inline src/base/Server.ts]: As above

File: src/base/Server.ts [hunk 5]
```diff
@@ -120,20 +120,26 @@ export default abstract class Server {
...
 
   /**
    * Run the server with graceful shutdown on `SIGINT` or `SIGTERM`
    */
   public async run(): Promise<void> {
     if (process !== undefined) {
-      const stop = async (): Promise<void> => {
-        await this.stop()
+      const stop = (): void => {
+        this.stop().catch(e => console.warn('Could not stop the server\n', e))
```
Comment (nokome) [inline src/base/Server.ts]: This would be better as `log.warn` with `const log = getLogger(`executa:base:server`) - for consistency.

**Comment by nokome** [comment]
Mmmmm, confusion much. I now see that rollup config etc is there. Must have been looking at just the files changes in a single commit previously. Shall continue playing....

**Comment by nokome** [comment]
@alex-ketch I've made some fixes and improvements so now the brower-based `Executor` can talk to the node-based `Executor` using either HTTP or Websocket :tada: . See `index.html`.

There are some optimizations that could be made to bundle size quite easily, including uglification and removing dependency on `Ajv` and `@stencila/schema` for browser based `Executors`. But IMO, that's not a priority right now. I think it makes sense to try to integrate this into `designa` first and see how the current API works when used in that context.

**Comment by alex-ketch** [comment]
Awesome job @nokome, gave it a spin and things look good 🎉 

Agreed about the following as well, so will merge.

> There are some optimizations that could be made to bundle size quite easily, including uglification and removing dependency on Ajv and @stencila/schema for browser based Executors. But IMO, that's not a priority right now. I think it makes sense to try to integrate this into designa first and see how the current API works when used in that context.

**Pull request merged**